### PR TITLE
Feature/string cache improvements

### PR DESCRIPTION
**Pull request by omerbenamram**
Apparently utilizing cached string yield only a modest performance increast (<5%).
It does make some flow significatly simpler as all seeks are eliminated while reading the binxml data.

### CODE REVIEW

File: src/string_cache.rs [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
 use crate::ChunkOffset;
 
+use log::trace;
+use std::borrow::BorrowMut;
 use std::collections::HashMap;
 use std::io::{Cursor, Seek, SeekFrom};
 
-pub type StringHash = u16;
+#[derive(Debug)]
+pub struct StringCache<'a>(HashMap<ChunkOffset, BinXmlName<'a>>);
```
Comment (ohadravid) [inline src/string_cache.rs]: Can the cache contain not-owned strings? Maybe we should simplify here with `HashMap<ChunkOffset, String>`?

File: src/string_cache.rs [hunk 2]
```diff
@@ -1,43 +1,46 @@
...
 
-pub type CachedString = (String, StringHash, ChunkOffset);
+impl<'a> StringCache<'a> {
+    pub fn populate(data: &'a [u8], offsets: &[ChunkOffset]) -> DeserializationResult<Self> {
+        let mut cache = HashMap::new();
+        let mut temp_cursor = Cursor::new(data);
```
Comment (ohadravid) [inline src/string_cache.rs]: Temp?

File: src/string_cache.rs (hunk 1 above)
Comment (omerbenamram) [reply src/string_cache.rs]: Done.

File: src/string_cache.rs (hunk 2 above)
Comment (omerbenamram) [reply src/string_cache.rs]: changed name

**Pull request merged**